dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/resourcetable/table/CategoryResourceTable.java

```diff
@@ -63,22 +65,27 @@ public ResourceTableType getTableType()
     @Override
     public String getCreateTempTableStatement()
     {
-        UniqueNameVerifier uniqueNameVerifier = new UniqueNameVerifier();
+        Set<String> usedColumnNames = new HashSet<>();
 
         String statement = "create table " + getTempTableName() + " (" +
             "categoryoptioncomboid bigint not null, " +
             "categoryoptioncomboname varchar(255), ";
 
         for ( Category category : objects )
         {
-            statement += uniqueNameVerifier.ensureUniqueShortName( category ) + " varchar(230), ";
+            statement += quote( category.getShortName() ) + " varchar(230), ";
             statement += quote( category.getUid() ) + " character(11), ";
+            usedColumnNames.add( category.getShortName() );
         }
 
         for ( CategoryOptionGroupSet groupSet : groupSets )
         {
-            statement += uniqueNameVerifier.ensureUniqueShortName( groupSet ) + " varchar(230), ";
+            String columnName = usedColumnNames.contains( groupSet.getShortName() )
```

Comment:
I see the need for checking uniqueness across categories and category option group sets. Ideally these should have been separate tables but too late for that.